source/grid/grid_tools.cc

```diff
@@ -2645,16 +2645,42 @@ namespace GridTools
     // Get cell weighting if a signal has been attached to the triangulation
     if (!triangulation.signals.cell_weight.empty())
       {
+        // If an underlying shared::Tria allows artificial cells,
+        // then save the current set of subdomain ids, and set
+        // subdomain ids to the "true" owner of each cell. we later
+        // restore these flags.
+        const parallel::shared::Triangulation<dim, spacedim> *shared_tria =
+          (dynamic_cast<const parallel::shared::Triangulation<dim, spacedim> *>(
+            &triangulation));
+        std::vector<types::subdomain_id> saved_subdomain_ids;
```

Comment:
switch the order of these two declarations -- you use `shared_tria` immediately after this line, so it's easier to read if you have a ABB pattern than a BAB pattern (A, B: declaration or use of a variable).